src/rules/memberAccessRule.ts

```diff
@@ -114,19 +121,24 @@ function walk(ctx: Lint.WalkContext<Options>) {
         }
     }
 
-    function check(node: ts.ClassElement): void {
+    function check(node: ts.ClassElement | ts.ParameterDeclaration): void {
         if (Lint.hasModifier(node.modifiers, ts.SyntaxKind.ProtectedKeyword, ts.SyntaxKind.PrivateKeyword)) {
             return;
         }
         const publicKeyword = getModifier(node, ts.SyntaxKind.PublicKeyword);
         if (noPublic && publicKeyword !== undefined) {
-            const start = publicKeyword.end - "public".length;
-            ctx.addFailure(
-                start,
-                publicKeyword.end,
-                Rule.FAILURE_STRING_NO_PUBLIC,
-                Lint.Replacement.deleteFromTo(start, getNextToken(publicKeyword, ctx.sourceFile)!.getStart(ctx.sourceFile)),
-            );
+            const readonlyKeyword = getModifier(node, ts.SyntaxKind.ReadonlyKeyword);
+            // public is not optional for parameter property without the readonly modifier
+            const isPublicOptional = !ts.isParameterPropertyDeclaration(node) || readonlyKeyword !== undefined;
```

Comment:
you don't need to check for parameter properties here again. If a `ts.ParameterDeclaration` gets passed to this function, it's always a parameter property.
You can replace the method call with `node.kind !== ts.SyntaxKind.Parameter`